pkg/spec/spec_test.go

```diff
@@ -0,0 +1,44 @@
+package spec
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"reflect"
+)
+
+func TestConflictingFields(t *testing.T) {
+	// This array should ideally go away and every struct in spec.go should
+	// get tested automatically, but for now, let's pass in all the structs
+	// https://stackoverflow.com/questions/20803758/how-to-get-all-defined-struct-in-golang
+	structsToTest := []interface{}{
+		&VolumeClaim{},
+		&ServicePortMod{},
+		&ServiceSpecMod{},
+		&IngressSpecMod{},
+		&EnvFromSource{},
+		&ConfigMapEnvSource{},
+		&Container{},
+		&ConfigMapMod{},
+		&PodSpecMod{},
+		&App{},
```

Comment:
Yep, the embedded structs will be taken care of, but not -
```golang
	VolumeClaims               []VolumeClaim     `json:"volumeClaims,omitempty"`
	ConfigMaps                 []ConfigMapMod    `json:"configMaps,omitempty"`
	Services                   []ServiceSpecMod  `json:"services,omitempty"`
	Ingresses                  []IngressSpecMod  `json:"ingresses,omitempty"`
```, but we also need to test those.

Ideally we should be getting all the structs in this file and test all one by one, and ignore testing if tested before, using ast or something, but that's for another PR sometime I guess.